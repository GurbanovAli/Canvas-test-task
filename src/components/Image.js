import React from 'react';


export default function Image(...args) {

   async function getImage(...args) { // function that returns background image
      const accessKey = 'a4e459094ff30eccca82da8258dd3e4e4ce923d584ab01bb47d1e4d364388591';
      const city = 'Minsk';
      const url = `https://api.unsplash.com/photos/random?query=${city}&client_id=${accessKey}`;
      const data = await fetch(url).then((res) => res.json());
      const root = document.getElementById('main');
      const img = root.style.backgroundImage = `url(${data.urls.full}&w=1100)`;
      return img;
   }

   return (
      <div>
      <button id = 'image'
      className = 'button download_button'
      onClick = { getImage } > Change Background </button>
      </div>
   );
}
