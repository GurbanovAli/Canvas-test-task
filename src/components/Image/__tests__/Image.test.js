import React from 'react';
import { render } from '@testing-library/react';
import Image from '../Image';
import Control from '../../ControlFile';


test('<Image /> spec', () => {
    const { container } = render(
      <Control>
        <Image />
      </Control>
    );
    expect(container).toMatchSnapshot();
  });
