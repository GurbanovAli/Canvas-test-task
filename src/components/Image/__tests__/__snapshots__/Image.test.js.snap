// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> spec 1`] = `
<div>
   
  <div>
    <button
      class="button download_button"
      id="image"
    >
       Change Background 
    </button>
  </div>
   
</div>
`;
